src/guide/project-maintenance.md

```diff
@@ -0,0 +1,113 @@
+
+### Performing the merge
+
+The clone we set up above is used each time you want to bring changes from mainline in to your project. This is done by first pulling the changes from mainline and then pushing to your personal fork. A regular PR will then be created which you can integrate into your main project repository. It sounds easy, and it is, but there are a few details below to keep in mind.
+
+~~~shell
+cd project-merge
+git pull upstream master
+git push
+git switch -c Merge_mainline
+git fetch mainline
+git merge mainline/master
+~~~
+
+We start by updating the local fork with the latest changes from the main project repository. Note that we then create a new branch in which the merge will happen.
+
+The commands above will likely run without a hitch up until the `git merge`. This is where you need to combine the changes that were made in mainline with the changes that have been made in your project repository. If there are no conflicts you're in luck, then the merge will be completely automated and you will end up with a committed merge. If there are conflicts however you'll need to manually go through the files where the conflicts are and make sure you select the correct version for each change. Using `git status` you can see what files that need to be merged. Depending on how much code your project has touched, this can be quite a bit of work.
+
+For complicated merges, see [Sharing the work](#sharing-the-work) below.
+
+### Test before push
+
+Regardless of if you encountered conflicts or not, you should always build and test your merge before pushing it to your project repository. Testing needs to be done even when there are no textual conflicts as changes like for instance a rename can result in a compile or test error without any conflict. One could argue that `git merge --no-commit` could be used and have logical errors fixed in the merge commit. However, a subsequent "Fix logical merge errors" commit, is in fact more useful, as it clearly shows the project specific adjustments needed for incoming changes.
+
+It's always okay to have further commits to clean up after a merge. Hiding a large amount of reworking project code to fit with upstream changes in a single merge commit will make it hard for further errors post integration to be identified.
+
+### The commit
+
+Once you have a working version of your merged code you're ready to create the merge commit and push. Please note that `git commit` is only needed if there were conflicts. If the changes were successfully merged by `git merge`, you already have a committed merge.
+
+~~~
+git commit -m "Merge"
+git push --set-upstream origin Merge_mainline
+~~~
+
+Now it's time to create the PR on GitHub. Just opening the PR page in your browser will most often be enough to see a message about new code that was pushed to your personal fork, with a question if you want to create a PR. You do.
```

Comment:
>You do.

I like the stylistic choice :-), but it might not be obvious to everyone what you mean and instead sound like an incomplete sentence. "Click the button to initiate the PR", or something of the sort, will be clearer.